scripts/release/prepare-release.sh

```diff
@@ -0,0 +1,370 @@
+    if [ -t 0 ]; then
+        # Interactive mode
+        read -r -e -p "New v1 version: " -i "${suggested_v1}" user_version_v1
+        
+        echo "Current v2 version: ${current_version_v2}"
+        read -r -e -p "New v2 version: " -i "${suggested_v2}" user_version_v2
+    else
+        # Non-interactive mode - use suggested versions
+        user_version_v1="${suggested_v1}"
+        user_version_v2="${suggested_v2}"
+        log_info "Non-interactive mode: using default versions v1=${user_version_v1}, v2=${user_version_v2}"
+    fi
+    
+    new_version_v1="v${user_version_v1}"
+    new_version_v2="v${user_version_v2}"
+    
+    log_success "Using new versions: v1=${new_version_v1}, v2=${new_version_v2}"
+}
+
+# Create release PR
+create_release_pr() {
+    log_info "Creating release PR..."
+    
+    # Create temporary changelog file
+    temp_changelog=$(mktemp)
+    make changelog > "$temp_changelog"
+    
+    # Update UI submodule to latest version
+    log_info "Updating UI submodule to latest version..."
+    if [ -d "jaeger-ui" ]; then
+        pushd jaeger-ui > /dev/null
+        git checkout main
+        git pull origin main
+        # Get the latest commit hash for the UI version
+        ui_latest_commit=$(git rev-parse HEAD)
+        ui_latest_tag=$(git describe --tags --abbrev=0 2>/dev/null || echo "latest")
+        popd > /dev/null
+        log_success "UI submodule updated to latest: ${ui_latest_tag} (${ui_latest_commit:0:8})"
+    else
+        log_warning "jaeger-ui submodule not found, skipping UI update"
+    fi
+
+    # Update CHANGELOG.md with new version header and generated content
+    log_info "Updating CHANGELOG.md..."
+    current_date=$(date +"%Y-%m-%d")
+    if [ -f "CHANGELOG.md" ]; then
+        temp_full_changelog=$(mktemp)
+        # Header matches guideline: "1.x.x / 2.x.x (YYYY-MM-DD)"
+        echo "${new_version_v1} / ${new_version_v2} (${current_date})" > "$temp_full_changelog"
+        echo "" >> "$temp_full_changelog"
+        cat "$temp_changelog" >> "$temp_full_changelog"
+        echo "" >> "$temp_full_changelog"
+        cat CHANGELOG.md >> "$temp_full_changelog"
+        mv "$temp_full_changelog" CHANGELOG.md
+        log_success "CHANGELOG.md updated"
+    else
+        log_error "CHANGELOG.md not found"
+        rm -f "$temp_changelog"
+        exit 1
+    fi
+
+    # Create PR title and body
+    pr_title="Prepare release ${new_version_v1} / ${new_version_v2}"
+    pr_body=$(cat << EOF
+## Release Preparation
+
+This PR automates the release preparation for ${new_version_v1} / ${new_version_v2}.
+
+### Changes Made:
+- [x] Updated CHANGELOG.md with new version section
+- [x] Generated changelog content using \`make changelog\`
+- [x] Updated UI submodule to latest version
+
+### Next Steps:
+1. Review and merge this PR
+2. Update UI submodule to latest version
+3. Create release tags: \`git tag ${new_version_v1} -s\` and \`git tag ${new_version_v2} -s\`
```

Comment:
these steps are already documented in the issue created by start-release, there's no need to repeat them here. The PR description should link back to the issue.